src/jwplayer/utils/useJwpWrapperInit.ts

```diff
@@ -0,0 +1,44 @@
+import { useEffect, useState } from 'react';
+import getSponsoredVideos from 'utils/getSponsoredVideos';
+import { recordVideoEvent, VIDEO_RECORD_EVENTS } from 'jwplayer/utils/videoTimingEvents';
+
+interface WindowWithSponsoredVideos extends Window {
+	sponsoredVideos?: string[];
+}
+
+declare let window: WindowWithSponsoredVideos;
+
+export default function useJwpWrapperInit(loadEmbed: () => void, shouldLoadSponsoredContentList: boolean) {
+	const [loadEmbedExecuted, setLoadEmbedExecuted] = useState(false);
+
+	useEffect(() => {
+		if (shouldLoadSponsoredContentList) {
+			const retrieveSponsoredVideo = async () => {
+				if (window?.sponsoredVideos?.length > 0) {
+					console.debug('sponsoredVideos already retrieved');
+					return;
+				}
+
+				const sponsoredVideoResponse = await getSponsoredVideos();
+				if (sponsoredVideoResponse && typeof window !== undefined) {
+					window.sponsoredVideos = sponsoredVideoResponse;
+					console.debug('Retrieved sponsoredVideos list');
+				} else {
+					console.debug(
+						'Could not set sponsored videos. Either window the fetched sponsoredVideo list were undefined.',
+					);
+				}
+			};
+			retrieveSponsoredVideo().catch((e) => {
+				console.error('There was an issue with retrieving Sponsored Videos. ', e);
+			});
+		} else {
+			console.debug('Loading of Sponsored Content Video List was disabled.');
+		}
+		recordVideoEvent(VIDEO_RECORD_EVENTS.JW_PLAYER_INIT_RENDER);
+		loadEmbed();
+		setLoadEmbedExecuted(true);
```

Comment:
I've just started learning React, so I apologize if my question seems... "basic" :D If the commented lines rely on the outcome of `retrieveSponsoredVideo`, will they function correctly? I recently read an [article about using async functions in watchEffect](https://devtrium.com/posts/async-functions-useeffect), where the author mentioned some important considerations.

Reply:
I'm also not a React expert 😊 Thanks for sharing the article! I give it a go and my understanding is that since we don't use the results from the async call in the state we don't need to worry here 🤔 The code has changed but indeed it could've been a source of race-condition 👍